src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -345,13 +347,17 @@ static void ClearStack(CountingConcurrentStack<T> stack)
 
         public void LowMemory(LowMemorySeverity lowMemorySeverity)
         {
-            if (lowMemorySeverity != LowMemorySeverity.ExtremelyLow)
-                return;
+            if (LowMemoryFlag.Raise())
+            {
+                Interlocked.Increment(ref _generation);
+            }
 
-            if (LowMemoryFlag.Raise() == false)
+            if (_isExtremelyLowMemory.Raise() == false || lowMemorySeverity != LowMemorySeverity.ExtremelyLow)
```

Comment:
You are *always* raising the extremely low memory flag here.

Reply:
done